### PR TITLE
Updated the FE to the latest version of the form

### PR DESCRIPTION
**Pull request by sirLisko**
**What**  
- fixed the API returning an error in case something goes wrong
- updated the FE to the latest version of the form
  - added missing step
  - changed copy
  - etc.
- fixed eligibility criteria answer check
- improved postcode lookup
- improved file upload
- improved summary step
- started aligning FE to BE
- added form hints
- removed unused components

### CODE REVIEW

File: components/FileUpload/FileUpload.jsx [hunk 1]
```diff
@@ -6,12 +6,13 @@ import DeleteIcon from 'components/Icons/DeleteIcon';
...
   label,
+  sublabel,
+  hint,
   name,
   fileList = [],
   onFileUploaded,
   onFileDelete,
-  uploadPrefix = 'test',
-  isSingleFile
+  uploadPrefix = 'test'
```
Comment (jbevan4) [inline components/FileUpload/FileUpload.jsx:15]: Do we still want this?

File: components/Steps/BusinessDetails.jsx [hunk 2]
```diff
@@ -51,38 +70,34 @@ const Step1 = props => {
...
       <TextInput
-        {...getInputProps('business', 'percentageFallIncome')}
+        {...getInputProps('business', 'councilRentAccountNumber')}
         type="number"
-        register={register({ min: 0, max: 100 })}
+        register={register}
       />
       <TextInput
-        {...getInputProps('business', 'businessPremises')}
+        {...getInputProps('business', 'numberEmployes')}
```
Comment (StephenThomson) [inline components/Steps/BusinessDetails.jsx:83]: *Employees

File: components/FileUpload/FileUpload.jsx (hunk 1 above)
Comment (sirLisko) [reply components/FileUpload/FileUpload.jsx:15]: it's `test` as a fallback, if it's not provided when invoking the function

File: components/Steps/EligibilityCriteria.jsx [hunk 3]
```diff
@@ -22,59 +24,96 @@ const Step1 = props => {
...
+            options={[
+              'Small or micro business in shared offices or workspaces without town business rates assessment',
+              'Small or micro businesses in the retail, hospitality and leisure sector with a rateable value of between £51,000-£60,000 (inclusive)',
+              'Small or micro business whose income and turnover is directly related to the Retail, Hospitality and Leisure sector',
+              'Regular market trader with fixed building costs',
+              'Bed & Breakfast who pay Council Tax',
+              'Charity properties who occupy a commercial property with a rateable value of £15,000 or less',
+              'Ofsted registered nurseries (not within a domestic premises)'
```
Comment (StephenThomson) [inline components/Steps/EligibilityCriteria.jsx:61]: Maybe as these are so long we could explain them first in a text box and then give shortened version in the select component

Comment (StephenThomson) [inline components/Steps/EligibilityCriteria.jsx:61]: Maybe as these are so long we could explain them first in a text box and then give shortened version in the select component

File: components/Steps/Summary.jsx [hunk 4]
```diff
@@ -18,12 +18,13 @@ const SummarySection = ({ formData, name, title, slug }) => (
...
-            : typeof value === 'boolean'
-            ? JSON.stringify(value)
-            : value
+        value: Array.isArray(value)
+          ? value.map(v => MultiValue(v.split('/').pop()))
+          : typeof value === 'object'
+          ? Object.values(value).map(MultiValue)
+          : typeof value === 'boolean'
+          ? JSON.stringify(value)
+          : value
```
Comment (StephenThomson) [inline components/Steps/Summary.jsx:27]: These nested turnaries are a bit hard to read.

File: components/FileUpload/FileUpload.jsx (hunk 1 above)
Comment (jbevan4) [reply components/FileUpload/FileUpload.jsx:15]: Ah okay no worries :) lgtm

File: pages/api/applications/index.js [hunk 5]
```diff
@@ -10,6 +10,6 @@ export default async (req, res) => {
     res.end(JSON.stringify(clientGeneratedId));
   } catch (error) {
     res.statusCode = 500;
-    res.end();
+    res.end(error.message);
```
Comment (StephenThomson) [inline pages/api/applications/index.js:13]: Do we need to JSON.stringify this?

File: components/FileUpload/FileUpload.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sirLisko) [reply components/FileUpload/FileUpload.jsx:15]: @jbevan4 spotted the same 😅 
> it's test as a fallback, if it's not provided when invoking the function

File: components/Steps/EligibilityCriteria.jsx (hunk 3 above)
Comment (sirLisko) [reply components/Steps/EligibilityCriteria.jsx:61]: Already highlighted to Gideon, he is looking into this 👍

File: components/Steps/Summary.jsx (hunk 4 above)
Comment (sirLisko) [reply components/Steps/Summary.jsx:27]: Do you think a `switch` would be more clear?

Comment (sirLisko) [reply components/Steps/Summary.jsx:27]: Merging this, if you think it's relevant I can create another PR :)

**Pull request merged**